ci-operator/step-registry/openstack/provision/dnsrecords/openstack-provision-dnsrecords-commands.sh

```diff
@@ -83,6 +83,28 @@ if [ -f "${SHARED_DIR}/MIRROR_REGISTRY_IP" ]; then
   cp "${TMP_DIR}/dns_mirror_registry.json" "${SHARED_DIR}/dns_up.json"
 fi
 
+if [ -f "${SHARED_DIR}/HIVE_FIP_API" ] && [ -f "${SHARED_DIR}/HIVE_FIP_INGRESS" ]; then
+  HIVE_FIP_API=$(<"${SHARED_DIR}"/HIVE_FIP_API)
+  HIVE_CLUSTER_NAME=$(<"${SHARED_DIR}"/HIVE_CLUSTER_NAME)
+  if [[ "${HIVE_FIP_API}" =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    HIVE_FIP_API_RECORD_TYPE="A"
+  else
+    HIVE_FIP_API_RECORD_TYPE="AAAA"
+  fi
+  echo "Creating Hive API DNS $HIVE_FIP_API_RECORD_TYPE record for ${HIVE_CLUSTER_NAME}.$BASE_DOMAIN"
+  jq '.Changes += [{"Action": "UPSERT", "ResourceRecordSet": {"Name": "api.'${HIVE_CLUSTER_NAME}'.'${BASE_DOMAIN}'.", "Type": "'${HIVE_FIP_API_RECORD_TYPE}'", "TTL": 300, "ResourceRecords": [{"Value": "'${HIVE_FIP_API}'"}]}}]' "${SHARED_DIR}/dns_up.json" > "${TMP_DIR}/dns_hive_api.json"
+  cp "${TMP_DIR}/dns_hive_api.json" "${SHARED_DIR}/dns_up.json"
+
+  HIVE_FIP_INGRESS=$(<"${SHARED_DIR}"/HIVE_FIP_INGRESS)
+  if [[ "${HIVE_FIP_INGRESS}" =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    HIVE_FIP_INGRESS_RECORD_TYPE="A"
+  else
+    HIVE_FIP_INGRESS_RECORD_TYPE="AAAA"
+  fi
+  echo "Creating Hive Ingress DNS $HIVE_FIP_INGRESS_RECORD_TYPE record for ${HIVE_CLUSTER_NAME}.$BASE_DOMAIN"
+  jq '.Changes += [{"Action": "UPSERT", "ResourceRecordSet": {"Name": "*.apps.'${HIVE_CLUSTER_NAME}'.'${BASE_DOMAIN}'.", "Type": "'${HIVE_FIP_INGRESS_RECORD_TYPE}'", "TTL": 300, "ResourceRecords": [{"Value": "'${HIVE_FIP_INGRESS}'"}]}}]' "${SHARED_DIR}/dns_up.json" > "${TMP_DIR}/dns_hive_ingress.json"
+  cp "${TMP_DIR}/dns_hive_ingress.json" "${SHARED_DIR}/dns_up.json"
```

Comment:
Won't this overwrite the file copied on L96? Should this instead be some variant of
```suggestion
  cat "${TMP_DIR}/dns_hive_ingress.json" >> "${SHARED_DIR}/dns_up.json"
```
?

Reply:
Hi @2uasimojo  Sorry for the late reply, and thank you for your reminder.

In the current implementation, each jq operation reads the latest ${SHARED_DIR}/dns_up.json and appends a new DNS record to the existing .Changes array. A temporary file is then generated and used to overwrite the original. So while it may appear to be an overwrite operation, each step is actually an incremental update based on the latest context, and no previous records are lost.
Using cat >> would directly append raw text, which could result in an invalid JSON structure.